go-controller/pkg/ovn/ovn.go

```diff
@@ -696,14 +699,33 @@ func (oc *Controller) WatchCRD() {
 			klog.Infof("Deleting CRD %s from cluster", crd.Name)
 			if crd.Name == egressfirewallCRD {
 				oc.egressFirewallDNS.Shutdown()
+				oc.deleteDefaultAllowACLsForEgressFirewall()
 				oc.watchFactory.RemoveEgressFirewallHandler(oc.egressFirewallHandler)
+				oc.watchFactory.RemoveNodeHandler(oc.egressFirewallNodeHandler)
 				oc.egressFirewallHandler = nil
+				oc.egressFirewallNodeHandler = nil
 				oc.watchFactory.ShutdownEgressFirewallWatchFactory()
 			}
 		},
 	}, nil)
 }
 
+// WatchEgressFirewallNodes starts the watching of nodes resource and calls
+// back the appropriate handler logic
+func (oc *Controller) WatchEgressFirewallNodes() *factory.Handler {
+	return oc.watchFactory.AddNodeHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			node := obj.(*kapi.Node)
+			oc.addNodeForEgressFirewall(node)
```

Comment:
I am not sure what the overhead is for handlers is but would it be better to check and add this funtion to add`addNode()` and the corresponding deletion in `deleteNode()`

Reply:
I only did it like this because we don't have a feature flag for egress firewall, and we shouldn't be creating all allow rules on cluster which don't use egress firewall.